src/app/Recordings/RecordingActions.tsx

```diff
@@ -58,18 +62,26 @@ export const RecordingActions: React.FC<RecordingActionsProps> = ({ recording, u
   }, [context.api, setGrafanaEnabled, addSubscription]);
 
   const grafanaUpload = React.useCallback(() => {
-    notifications.info('Upload Started', `Recording "${recording.name}" uploading...`);
-    uploadFn();
-    addSubscription(
-      context.notificationChannel.messages(NotificationCategory.GrafanaUploadSuccess).subscribe(() => {
-        notifications.success('Upload Success', `Recording "${recording.name}" uploaded`);
-        context.api
-          .grafanaDashboardUrl()
-          .pipe(first())
-          .subscribe((url) => window.open(url, '_blank'));
-      }),
-    );
-  }, [addSubscription, notifications, context.api, context.notificationChannel, recording, uploadFn]);
+    uploadFn().subscribe((jobId) => {
```

Comment:
This `.subscribe()` creates a `Subscription`, which should be tracked with `addSubscription`.

Maybe this subscription can be mapped so that the later inner notification subscription and this one are combined into one? I think that might also remove the need to track the `jobIds` in a separate array data structure.

Reply:
> Maybe this subscription can be mapped so that the later inner notification subscription and this one are combined into one? I think that might also remove the need to track the jobIds in a separate array data structure.

What would this look like? AutomatedAnalysisCard and utils.tsx both seem to have nested subscriptions like this.

Reply:
I'm imagining something like this:

```
const sub = uploadFn.pipe(
  tap(() => notifications.info('Upload Started', `Recording "${recording.name}" uploading...`)),
  concatMap(jobId=>
    context.notificationChannel.messages(NotificationCategory.GrafanaUploadSuccess.pipe(filter(n => n.message.jobId === jobId))
  ),
  first(),
  tap(() => notifications.success('Upload Success', `Recording ${recording.name} uploaded`)),
  concatMap(() => context.api.grafanaDashboardUrl().pipe(first()))
).subscribe((url) => window.open(url, '_blank'));
addSubscription(sub);
```

I might not have got all the `concatMaps` and things right, but something like that.